sofar/sofar.py

```diff
@@ -0,0 +1,2125 @@
+                setattr(sofa, var.replace(".", "_"), value)
+            else:
+                dimensions = "".join([d for d in file[var].dimensions])
+                # SOFA only uses dtypes 'double' and 'S1' but netCDF has more
+                dtype = "string" if file[var].datatype == "S1" else "double"
+                _add_custom_api_entry(sofa, var.replace(".", "_"), value, None,
+                                      dimensions, dtype)
+                custom.append(var.replace(".", "_"))
+
+            # load variable attributes
+            for attr in [a for a in file[var].ncattrs() if a not in skip]:
+
+                value = getattr(file[var], attr)
+
+                if not hasattr(sofa, var.replace(".", "_") + "_" + attr):
+                    _add_custom_api_entry(
+                        sofa, var.replace(".", "_") + "_" + attr, value, None,
+                        None, "attribute")
+                    custom.append(var.replace(".", "_") + "_" + attr)
+                else:
+                    setattr(sofa, var.replace(".", "_") + "_" + attr, value)
+
+        # do not allow writing read only attributes any more
+        sofa._protected = True
+
+    # notice about custom entries
+    if custom:
+        warnings.warn(("SOFA file contained custom entries "
+                      f"({', '.join(custom)})"))
+
+    # update api
+    if verify:
+        try:
+            sofa.verify(version)
+        except: # noqa (No error handling - just improved verbosity)
+            raise ValueError((
+                "The SOFA object could not be verified, maybe do to errornous "
+                "data. Call sofa=sofar.read_sofa(filename, verify=False) and "
+                "than sofa.verify() to get more information"))
+
+    return sofa
+
+
+def write_sofa(filename: str, sofa: Sofa, version="latest", compression=9):
+    """
+    Write a SOFA object to disk as a SOFA file.
+
+    Parameters
+    ----------
+    filename : str
+        The filename. '.sofa' is appended to the filename, if it is not
+        explicitly given.
+    sofa : object
+        The SOFA object that is written to disk
+    version : str
+        The SOFA object is verified and updated with :py:func:`~Sofa.verify`
+        before writing to disk. Version specifies, which version of the
+        convention is used:
+
+        ``'latest'``
+            Use the latest version upgrade the SOFA file if required.
+        ``'match'``
+            Match the version of the SOFA object.
+        str
+            Version string, e.g., ``'1.0'``.
+
+        The default is ``'latest'``.
+    compression : int
+        The level of compression with ``0`` being no compression and ``9``
+        being the best compression. The default of ``9`` optimizes the file
+        size but increases the time for writing files to disk.
+    """
+
+    # check the filename
+    if not filename.lower().endswith('.sofa'):
+        filename += ".sofa"
```

Comment:
same comment as above, might often result in things like `file.soaf.sofa`
I'd go for raising an error if the `.sofa` file extension is missing

Reply:
yes - see above